Solutions/CohesitySecurity/Tools/grant_keyvault_permission.sh

```diff
@@ -0,0 +1,42 @@
+#!/bin/zsh
+SCRIPT=$(realpath "$0")
+SCRIPTPATH=$(dirname "$SCRIPT")
+cd "$SCRIPTPATH"
+
+# Description: This script grants "Get" permissions to the specified user and playbooks on the Key Vault.
+
+# Set variables
+keyvault_prefix="cohesitypro"
+playbook_names=("Cohesity_Close_Helios_Incident" "Cohesity_CreateOrUpdate_ServiceNow_Incident" "Cohesity_Delete_Incident_Blobs" "Cohesity_Restore_From_Last_Snapshot" "Cohesity_Send_Incident_Email")
+user_object_id="142355ec-a29b-40b7-81c3-e769c76b1756"
```

Comment:
What's user_object_id and why is it hardcoded?

Reply:
we have to introduce some hardcode, bellowing is the user info for 142355ec:
{
    "businessPhones": [],
    "displayName": "Ravi Varanasi",
    "givenName": "Ravi",
    "id": "142355ec-a29b-40b7-81c3-e769c76b1756",
    "jobTitle": null,
    "mail": null,
    "mobilePhone": null,
    "officeLocation": null,
    "preferredLanguage": "en",
    "surname": "Varanasi",
    "userPrincipalName": "cohesity-siem_outlook.com#EXT#@cohesitysiemoutlook.onmicrosoft.com"
}
just moved the user_object_id into cohesity.json file

Reply:
1. Can't we get this information from an 'az cli' command?
2. This automation should be usable from any account. For example, I don't use Ravi's account.
3. Why does a user need the 'Get' privilege on Key vault? Could you please mention the workflow that requires the user to have this privilege? And the documentation in our code which says that the user requires this privelege?

Reply:
https://learn.microsoft.com/en-us/azure/role-based-access-control/role-assignments-rest

https://learn.microsoft.com/en-us/azure/role-based-access-control/role-assignments-cli

Reply:
Yes or no, to obtain this information from the CLI, we need to search using either "Ravi Varanasi" or "cohesity-siem_outlook.com#EXT#@cohesitysiemoutlook.onmicrosoft.com". However, we still require some hardcoded input for the search. Among these options, the ID is more likely to be unique for identifying a user.

If you decide to run the automation from a different account, simply update the cohesity.json file to use the other user's ID.